api/v1alpha1/httproute_types.go

```diff
@@ -263,12 +266,27 @@ type ForwardToTarget struct {
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+	TargetPort *TargetPort `json:"targetPort,omitempty" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// Weight specifies the proportion of traffic to be forwarded to a targetRef.
+	// A valid weight value is 0-100. The sum of weights across targetRefs must
+	// equal 100. If only one targetRef is specified, weight is ignored and all
```

Comment:
Another option is that instead of weight specifying the fraction, the fraction of traffic to be forwarded to a targetRef is computed as weight / (sum of all weights in targetRefs). Then the constraint of "A valid weight value is 0-100. The sum of weights across targetRefs must equal 100" can be removed.

Reply:
@yiyangy thanks for the review. I've updated the godocs based on your feedback.